File: backend/app/schemas/audit.py
from datetime import datetime

from pydantic import BaseModel


class AuditLogRead(BaseModel):
    id: str
    workspace_id: str
    actor_member_id: str | None = None
    action: str
    resource_type: str
    resource_id: str | None = None
    request_method: str | None = None
    request_path: str | None = None
    status: str
    ip_address: str | None = None
    user_agent: str | None = None
    metadata_json: dict[str, object]
    created_at: datetime

File: frontend/src/api/audit.ts
import { apiGet } from './client'

export interface AuditLog {
  id: string
  workspace_id: string
  actor_member_id?: string | null
  action: string
  resource_type: string
  resource_id?: string | null
  request_method?: string | null
  request_path?: string | null
  status: string
  ip_address?: string | null
  user_agent?: string | null
  metadata_json: Record<string, unknown>
  created_at: string
}

export interface AuditLogQuery {
  limit?: number
  action?: string
  resource_type?: string
  status?: string
  keyword?: string
}

export function listAuditLogs(query: AuditLogQuery = {}) {
  const params = new URLSearchParams()
  params.set('limit', String(query.limit ?? 50))
  if (query.action) params.set('action', query.action)
  if (query.resource_type) params.set('resource_type', query.resource_type)
  if (query.status) params.set('status', query.status)
  if (query.keyword) params.set('keyword', query.keyword)
  return apiGet<AuditLog[]>(`/audit/logs?${params.toString()}`)
}
